esphome/components/mcp3428/mcp3428.cpp

```diff
@@ -0,0 +1,168 @@
+#include "mcp3428.h"
+#include "esphome/core/hal.h"
+#include "esphome/core/log.h"
+
+namespace esphome {
+namespace mcp3428 {
+
+static const char *const TAG = "mcp3426/7/8";
+
+void MCP3428Component::setup() {
+  ESP_LOGCONFIG(TAG, "Setting up MCP3426/7/8...");
+  uint8_t anwser[3];
+  if (this->read(anwser, 3) != i2c::ErrorCode::NO_ERROR) {
+    this->mark_failed();
+    return;
+  }
+
+  ESP_LOGCONFIG(TAG, "Configuring MCP3426/7/8...");
+
+  /* config byte structure: (bit|description)
+   * 7 | ready bit, 0 means new data in the result
+   * 6 | Channel selection bit 1
+   * 5 | Channel selection bit 0
+   * 4 | conversion mode bit (1 continuous mode, 0 singe shot)
+   * 3 | Resolution bit 1
+   * 2 | Resolution bit 0
+   * 1 | Gain selection bit 1
+   * 0 | Gain selection bit 0
+   */
+  uint8_t config = 0;
+  // set ready bit and conversion mode bit
+  //                  0b0xx0xxxx
+  if (this->continuous_mode_) {
+    // initial state should be no new measurement in continuous mode
+    config = config | 0b10010000;
+  } else {
+    // don't initiate measurement in single shot mode
+    config = config & 0b01101111;
+  }
+  // leave channel at 1, gain at 1x, and resolution at 12 bit
+
+  if (this->write(&config, 1) != i2c::ErrorCode::NO_ERROR) {
+    this->mark_failed();
+    return;
+  }
+  this->prev_config_ = config;
+  single_measurement_active_ = false;
```

Comment:
Please make sure to always prefix member variables and functions with `this->`
```suggestion
  this->single_measurement_active_ = false;
```